gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerSessionIntegrateTest.java

```diff
@@ -320,39 +320,47 @@ public void shouldHaveTheSessionTimeout() throws Exception {
             cluster.close();
         }
 
-        // there will be on for the timeout and a second for closing the cluster
+        this.stopServer();
```

Comment:
was there a reason you had to `stopServer()` here to get this test to pass?

Reply:
Good catch. This was due to a missing piece of functionality in the new implementation.

[1] https://github.com/apache/tinkerpop/blob/master/gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Connection.java#L313

Added that now and tests passes without having to stop the server to close the session.